src/errors.rs

```diff
@@ -37,13 +47,17 @@ pub(crate) struct Error {
 }
 
 impl Error {
-    pub fn new<P: Into<PathBuf>>(source: io::Error, kind: ErrorKind, path: P) -> io::Error {
+    pub fn new(source: io::Error, kind: ErrorKind, path: PathBuf) -> io::Error {
+        io::Error::new(source.kind(), Self { kind, source, path })
+    }
+
+    pub fn new_from_ref(source: io::Error, kind: ErrorKind, path: &Path) -> io::Error {
```

Comment:
Wouldn't something like this work?

```
    pub fn new(source: io::Error, kind: ErrorKind, path: impl Into<PathBuf>) -> io::Error {
        Self::new(source, kind, path.into())
    }

    pub fn new(source: io::Error, kind: ErrorKind, path: PathBuf) -> io::Error {
        io::Error::new(source.kind(), Self { kind, source, path })
    }
```

Reply:
Sure and I don't think you need the `_new` either. I mentioned that in the OP:
>I made the crate's Error constructors non-generic so that rustc would tell me about all the callsites that needed attention.

There's no need to keep both `new` and `new_from_ref`.

Reply:
That said I do like the explicitness. It means you couldn't accidentally pass an `&PathBuf` and cause it to compile a new variant needlessly. It really should be either `&Path` or `PathBuf` and nothing else.
`new_from_ref` is more common than `new` and should therefore have a shorter name.